tests/python_package_test/test_callback.py

```diff
@@ -22,13 +22,7 @@ def test_early_stopping_callback_is_picklable(serializer):
 
 
 def test_early_stopping_callback_rejects_invalid_stopping_rounds_with_informative_errors():
-    with pytest.raises(ValueError, match="stopping_rounds should be an integer and greater than 0. got: 0"):
-        lgb.early_stopping(stopping_rounds=0)
-
-    with pytest.raises(ValueError, match="stopping_rounds should be an integer and greater than 0. got: -1"):
-        lgb.early_stopping(stopping_rounds=-1)
-
-    with pytest.raises(ValueError, match="stopping_rounds should be an integer and greater than 0. got: neverrrr"):
+    with pytest.raises(TypeError, match="early_stopping_round should be an integer. Got " "<class 'str'>"):
         lgb.early_stopping(stopping_rounds="neverrrr")
```

Comment:
Could you please add one more test here, confirming that passing an int `<= 0` into the callback results in its `enabled` attribute being set to `False`?

Something like this

```python
@pytest.mark.parametrize("early_stopping_round", [-10, -1, 0])
def test_early_stopping_callback_accepts_non_positive_stopping_rounds(stopping_rounds):
    cb = lgb.early_stopping(stopping_rounds=stopping_rounds)
    assert cb.enabled is False
```

That behavior isn't technically covered by the tests in `test_engine.py`, since the changes you've made to `train()` and `cv()` prevent `lgb.early_stopping()` from being called at all.

Reply:
Looks like you added a test for this in https://github.com/microsoft/LightGBM/pull/6406/commits/260fb9483910f7e52d15365b39448d5adef5ba8c. Thank you!